macro/src/span.rs

```diff
@@ -1,48 +1,19 @@
-use self::hash::Hasher;
 use proc_macro::Span;
 use std::env;
-use std::hash::{Hash, Hasher as _};
 
-#[cfg(feature = "sha2")]
-mod hash {
-    use sha2::Digest;
-
-    pub struct Hasher(sha2::Sha256);
-
-    impl Hasher {
-        pub fn new() -> Self {
-            Hasher(Digest::new())
-        }
-
-        fn finalize(&self) -> [u8; 32] {
-            self.0.clone().finalize().into()
-        }
-
-        pub fn finish_wide(self) -> u128 {
-            let [a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, ..] = self.finalize();
-            u128::from_ne_bytes([a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p])
-        }
-    }
-
-    impl std::hash::Hasher for Hasher {
-        fn finish(&self) -> u64 {
-            let [a, b, c, d, e, f, g, h, ..] = self.finalize();
-            u64::from_ne_bytes([a, b, c, d, e, f, g, h])
-        }
-
-        fn write(&mut self, bytes: &[u8]) {
-            self.0.update(bytes);
+use lazy_static::lazy_static;
+use tiny_keccak::{Hasher, Sha3};
+
+lazy_static! {
+    static ref SEED: Vec<u8> = {
+        if let Some(value) = env::var_os("CONST_RANDOM_SEED") {
```

Comment:
You would need to either use [`proc_macro::tracked_env::var`](https://doc.rust-lang.org/nightly/proc_macro/tracked_env/fn.var.html) (which is unstable) or `option_env!` to ensure that changing `CONST_RANDOM_SEED` causes a rebuild.

Reply:
Done